jerry-ext/args/arg-transform-functions.c

```diff
@@ -0,0 +1,249 @@
+  }
+
+  return jerry_create_undefined ();
+}
+
+static jerry_value_t
+jerryx_arg_transform_string_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  jerry_value_t to_string = jerry_value_to_string (js_arg);
+
+  if (jerry_value_has_error_flag (to_string))
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It can not be converted to a string.");
+  }
+
+
+  jerry_length_t dest_len = (jerry_length_t) iterator_p->c_arg_p->extra_info;
+  char *dest = jerryx_arg_iterator_get_dest (iterator_p);
+
+  jerry_size_t size = jerry_string_to_char_buffer (to_string,
+                                                   (jerry_char_t *) dest,
+                                                   dest_len);
+
+  if (size == 0)
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The size of the buffer is not large enough.");
+  }
+
+  jerry_release_value (to_string);
+
+  return jerry_create_undefined ();
+}
+
+/* function transform functions. */
+
+static jerry_value_t
+jerryx_arg_transform_function_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_function (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a function.");
+  }
+
+  jerry_value_t *func_p = jerryx_arg_iterator_get_dest (iterator_p);
+  *func_p = jerry_acquire_value (js_arg);
+
+  return jerry_create_undefined ();
+}
+
+/* Object's native pointer transform function */
+
+static jerry_value_t
+jerryx_arg_transform_native_pointer_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_object (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a object.");
+  }
+
+  const jerry_object_native_info_t *expected_info_p;
+  const jerry_object_native_info_t *out_info_p;
+  expected_info_p = (const jerry_object_native_info_t *) iterator_p->c_arg_p->extra_info;
+  void **ptr_p = (void **) jerryx_arg_iterator_get_dest (iterator_p);
+  bool is_ok = jerry_get_object_native_pointer (js_arg, ptr_p, &out_info_p);
+
+  if (!is_ok || out_info_p != expected_info_p)
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The object have no native pointer or type is not matched.");
+  }
+
+  return jerry_create_undefined ();
+}
+
+/* Ignore transform function */
+
+jerry_value_t
+jerryx_arg_transform_ignore (jerryx_arg_iterator_t *iterator_p)
+{
+  jerryx_arg_iterator_pop (iterator_p);
+  return jerry_create_undefined ();
+}
+
+#define JERRYX_ARG_TRANSFORM_DEF(type) jerry_value_t \
```

Comment:
We need 2 transform function for each type, required and optional. so the JERRYX_ARG_TRANSFORM_DEF macro will define both functions.